app/src/androidTest/java/ch/epfl/sdp/blindly/main_screen/ProfilePageTest.kt

```diff
@@ -73,10 +81,99 @@ class ProfilePageTest {
     }
 
+    @Test
+    fun settingsButtonRemovesAudioPlayerFragment() {
+        onView(withId(R.id.play_audio_profile_button)).perform(click())
+        onView(withId(R.id.settings_profile_button)).perform(click())
+        val audioPlayerFragment = getAudioPlayerFragment()
+        assert(audioPlayerFragment == null)
+    }
+
+    @Test
+    fun playAudioButtonCreatesAudioPlayerFragment() {
+        //Create and show the audio player
+        onView(withId(R.id.play_audio_profile_button)).perform(click())
+        val audioPlayerFragment = getAudioPlayerFragment()
+        assert(audioPlayerFragment != null)
+        if (audioPlayerFragment != null) {
+            assert(audioPlayerFragment.isVisible)
+        }
+    }
+
+    @Test
+    fun audioPlayerNotRemovedIfNotCreated() {
+        val audioPlayerFragment = getAudioPlayerFragment()
+        onView(withId(R.id.profile_relativeLayout)).perform(click())
+        assert(audioPlayerFragment == null)
+    }
+
+    @Test
+    fun clickingOutsideAudioPlayerFragmentRemovesIt() {
+        //Create and show the audio player
+        onView(withId(R.id.play_audio_profile_button)).perform(click())
+        //Remove the audio player
+        onView(withId(R.id.profile_relativeLayout)).perform(click())
+
+        assert(getAudioPlayerFragment() == null)
+    }
+
+    @Test
+    fun playAudioButtonCreateAudioPlayerFragment() {
+        //Create and show the audio player
+        onView(withId(R.id.play_audio_profile_button)).perform(click())
+        //Remove the audio player
+        onView(withId(R.id.profile_relativeLayout)).perform(click())
+        //Create it again
+        onView(withId(R.id.play_audio_profile_button)).perform(click())
+
+        val audioPlayerFragment = getAudioPlayerFragment()
+        assert(audioPlayerFragment != null)
+        if (audioPlayerFragment != null) {
+            assert(audioPlayerFragment.isVisible)
+        }
+
+    }
+
+    //TODO not yet implemented
+    /*@Test
+    fun recordButtonFiresRecordingActivity() {
+        //Create and show the audio player
+        onView(withId(R.id.play_audio_profile_button)).perform(click())
+
+        onView(withId(R.id.record_button)).perform(click())
+        intended(hasComponent(RecordingActivity::class.java.name))
+    }
+
+    @Test
+    fun recordButtonRemovedAudioPlayerFragment() {
+        //Create and show the audio player
+        onView(withId(R.id.play_audio_profile_button)).perform(click())
+
+        onView(withId(R.id.record_button)).perform(click())
+        val fragmentManager = fragment.childFragmentManager
+        val audioFragment = fragmentManager.findFragmentById(R.id.fragment_audio_container_view)
+        assert(audioFragment == null)
+    }
+
+     */
+
+    private fun getAudioPlayerFragment(): Fragment? {
+        val fragmentManager = fragment.childFragmentManager
+        return fragmentManager.findFragmentById(R.id.fragment_audio_container_view)
+    }
+
```

Comment:
I didn't repeat the comment for all tests, but I think it could be a good idea to change them all